server/proxy/pf_rdpgfx.c

```diff
@@ -147,9 +147,61 @@ static UINT proxy_MapSurfaceToWindow(RdpgfxClientContext* context,
 	return server->MapSurfaceToWindow(server, surfaceToWindow);
 }
 
+static UINT proxy_OnOpen(RdpgfxClientContext* context, BOOL *do_caps_advertise)
+{
+	WLog_DBG(TAG, "OnOpen");
+	RdpgfxServerContext* server = (RdpgfxServerContext*) context->custom;
+
+	if (NULL != do_caps_advertise)
+		*do_caps_advertise = FALSE;
+
+	printf("pf_rdpgfx: proxy_OnOpen: calling server's Open()\n");
+	// TODO: check if proxy server's dynvc is in ready state
+	// and that client supports gfx, else deinit the channel.
+
+	// we do this error check since the server's API doesn't return WTSAPI error codes
+	if (server->Open(server))
+	{
+		return CHANNEL_RC_OK;
+	}
+	return CHANNEL_RC_INITIALIZATION_ERROR;
+
+}
+
+static UINT proxy_OnClose(RdpgfxClientContext* context)
+{
+	WLog_DBG(TAG, "OnClose");
+	RdpgfxServerContext* server = (RdpgfxServerContext*) context->custom;
+
+	printf("pf_rdpgfx: proxy_OnClose: calling server's Close()\n");
+	return server->Close(server);
+}
+
+static UINT proxy_CapsConfirm(RdpgfxClientContext* context,
+	RDPGFX_CAPS_CONFIRM_PDU *capsConfirm)
+{
+	WLog_DBG(TAG, "CapsConfirm");
+	RdpgfxServerContext* server = (RdpgfxServerContext*) context->custom;
+
+	printf("proxy_CapsConfirm: calling server's CapsConfirm()\n");
+	return server->CapsConfirm(server, capsConfirm);
+}
+
+// called when proxy server receives `CapsAdvertise`
```

Comment:
Fix this comment (I think it should be more indicative). Also, use /**/ for comments